app/views/projects/attachments/index.html.erb

```diff
@@ -0,0 +1,52 @@
+<%= turbo_refreshes_with method: :morph, scroll: :preserve %>
+
+<%= turbo_frame_tag "attachment_modal" %>
+
+<div class="fixed-table-component">
+  <%= render Viral::PageHeaderComponent.new(
+    title: t(".title"),
+    subtitle: t(".subtitle", project_name: @project.name),
+  ) do |component| %>
+    <%= component.with_buttons do %>
+    <% if allowed_to?(:create_attachment?, @project) %>
+      <%= link_to t(".upload_files"),
+          new_namespace_project_attachment_path(id: @project.id),
+          data: {
+            turbo_frame: "attachment_modal",
+            turbo_stream: true,
+          },
+          class: "button button--size-default button--state-default" %>
+      <% end %>
+    <% end %>
+  <% end %>
+
+  <div class="flow-root">
+    <div class="flex flex-row-reverse items-center mb-4 space-x-2">
+      <%= search_form_for @q, url: namespace_project_attachments_path(@project.parent, @project), html: { "data-controller": "filters", "data-turbo-permanent": "true" } do |f| %>
+        <%= f.hidden_field :format, value: "turbo_stream" %>
+        <%= f.label :name_cont, "SEARCH", class: "sr-only" %>
+        <div class="relative lg:w-72">
+          <div
+            class="
+              absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none
+            "
+          >
+            <%= viral_icon(name: "magnifying_glass", classes: "h-5 w-5") %>
+          </div>
+          <%= f.search_field :puid_or_file_blob_filename_cont,
```

Comment:
After needing way too much help from @ericenns, added in 9f6ec81697b0b3f3d6ac274bffa8713633052f41.

Because of the current state of `project attachments`, pagy limit component `hrefs` needs to contain the entire url in order to preserve any sorting/filtering

Reply:
Nice work, I too find that stuff confusing.